src/basic/stat-util.c

```diff
@@ -365,7 +365,6 @@ bool stat_inode_same(const struct stat *a, const struct stat *b) {
          * a thorough check, comparing inode nr, backing device and if the inode is still of the same type. */
 
         return a && b &&
-                (a->st_mode & S_IFMT) != 0 && /* We use the check for .st_mode if the structure was ever initialized */
```

Comment:
i talked to christian, and we came to the conclusion that st_dev being zero can never be possible, hence can we replace the check with an st_dev nonzero check?